libs/ledger-live-common/src/families/canton/react.ts

```diff
@@ -0,0 +1,73 @@
+import { useCallback } from "react";
+import { getCurrencyBridge } from "../../bridge";
+import { CantonCurrencyBridge } from "@ledgerhq/coin-canton/types";
+import { CryptoCurrency } from "@ledgerhq/types-cryptoassets";
+import { Account } from "@ledgerhq/types-live";
+
+export type UseCantonAcceptOrRejectOfferOptions = {
+  currency: CryptoCurrency;
+  account: Account;
+  partyId: string;
+};
+
+export type TransferInstructionParams = {
+  contractId: string;
+  deviceId: string;
+  reason?: string;
+};
+
+export type TransferInstructionType =
+  | "accept-transfer-instruction"
+  | "reject-transfer-instruction"
+  | "withdraw-transfer-instruction";
+
+export function useCantonAcceptOrRejectOffer({
+  currency,
+  account,
+  partyId,
+}: UseCantonAcceptOrRejectOfferOptions) {
+  const cantonBridge = getCurrencyBridge(currency) as CantonCurrencyBridge;
+
+  const transferInstruction = useCallback(
+    (
+      { contractId, deviceId, reason }: TransferInstructionParams,
+      type: TransferInstructionType,
+    ) => {
+      return cantonBridge.transferInstruction(
+        currency,
+        deviceId,
+        account,
+        partyId,
+        contractId,
+        type,
+        reason,
+      );
+    },
+    [cantonBridge, currency, account, partyId],
+  );
+
+  // Wrapper functions for backward compatibility
+  const acceptOffer = useCallback(
```

Comment:
don't think we need the`useCallback` here, whenever the the component rerenders is because some property was updated, which will make `transferInstruction` being updated also, because the list of dependencies is the same as the component.